src/vse_sync_pp/analyze.py

```diff
@@ -0,0 +1,75 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+import sys
+import json
+import pandas as pd
+from argparse import ArgumentParser
+
+from .common import open_input
+
+from .parse import PARSERS
+from .analyzers import ppsdpll
+from . plot import Plotter
+
+ANALYZERS = {
+    cls.id_: cls for cls in (
+        ppsdpll.PhaseOffsetTimeError,
+        ppsdpll.ConstantPhaseOffsetTimeError,
+        ppsdpll.PhaseOffsetTimeDeviation,
+        ppsdpll.MaxPhaseOffsetTimeIntervalError
+    )
+}
+
+def main():
+    """Print Synchronization KPIs from parsed data to stdout"""
+    aparser = ArgumentParser(description=main.__doc__)
+    aparser.add_argument(
+        '-c', '--canonical', action='store_true',
+        help="parse canonical log data from input",
+    )
+    aparser.add_argument(
+        'input',
+        help="data parsed",
+    )
+    aparser.add_argument(
+        'parser', choices=tuple(PARSERS),
+        help="data to parse from the log file",
+    )
+    aparser.add_argument(
+        'analyzer', choices=tuple(ANALYZERS),
+        help="analyzers to run",
+        #nargs='+'
+    )
+    aparser.add_argument(
+        'output',
+        help="filename to output image to",
+    )
+    args = aparser.parse_args()
+
+    samples = pd.read_csv(args.input)
+    parser = PARSERS[args.parser]()
+    print(args.analyzer)
+    analyzer = ANALYZERS[args.analyzer]()
+    with open_input(args.input) as fid:
+        analyze = ppsdpll.Analyzer()
+        method = parser.canonical if args.canonical else parser.parse
+        for parsed in method(fid):
+            #print(parsed)
+            analyze.append(parsed)
+        analyzed = analyzer.calculate(analyze.retrieve(),"PRTC-B")
+    plotter  = Plotter("phaseoffset")
+    for analyzed_sample in analyze.transform():
+      plotter.append(analyzed_sample)
+    plotter.plot(args.output)
+
+    #for metric in args.analyzer:
```

Comment:
Propose we delete this for now: focus on the case of single analyzer before we get complicated.

Reply:
Agree.